cmd/parca-agent/main.go

```diff
@@ -364,9 +365,15 @@ func main() {
 	// Run group for target manager
 	{
 		ctx, cancel := context.WithCancel(ctx)
+		counter := promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "parca_agent_profiler_missing_stacks_total",
+				Help: "Number of missing profile stacks",
+			},
+			[]string{"type"})
 		g.Add(func() error {
 			level.Debug(logger).Log("msg", "starting target manager")
-			return tm.Run(ctx, m.SyncCh())
+			return tm.Run(ctx, m.SyncCh(), counter)
```

Comment:
I would rather see these types of dependencies pass through constructors and be an instance member of the type.

Reply:
agreed, this was entirely a hack to  see if it works, if the approach seems correct, there's some refactoring and cleanup that needs to be done.